forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -219,18 +249,63 @@ pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
                 bail!("Both `--salt` and `--default-salt` were specified: must choose one")
             }
         };
-        let contract_id = deploy_pkg(
-            &command,
-            pkg,
-            salt,
-            &provider,
-            &signing_key,
-            node_url.clone(),
-        )
-        .await?;
-        contract_ids.push(contract_id);
+        let deployed_contract_id = deploy_pkg(&command, pkg, salt, &provider, &signing_key).await?;
+
+        let proxy_id = match &pkg.descriptor.manifest_file.proxy {
+            Some(forc_pkg::manifest::Proxy {
+                enabled: true,
+                address: Some(proxy_addr),
+            }) => {
+                // Make a call into the contract to update impl contract address to 'deployed_contract'.
+
+                // Create a contract instance for the proxy contract using default proxy contract abi and
+                // specified address.
+                let proxy_contract =
+                    ContractId::from_str(proxy_addr).map_err(|e| anyhow::anyhow!(e))?;
+
+                update_proxy_contract_target(
+                    &provider,
+                    signing_key,
+                    proxy_contract,
+                    deployed_contract_id,
+                )
+                .await?;
+                Some(proxy_contract)
+            }
+            Some(forc_pkg::manifest::Proxy {
+                enabled: true,
+                address: None,
+            }) => {
+                // Deploy a new proxy contract.
+                let deployed_proxy_contract = deploy_new_proxy(
+                    pkg,
+                    &deployed_contract_id,
+                    &build_opts,
+                    &command,
+                    salt,
```

Comment:
Do we need to calculate the salt again inside of `deploy_new_proxy` rather than reusing the salt from the main contract?

Reply:
Hmm I don't think salt is something we calculate. I believe it is either provided by the user or we generate it randomly, basically something (32 byte num) we use to differentiate different contracts. It is not based on contract itself. We can regenerate a new random salt for proxy but i am not sure what it would add

Reply:
Okay, that makes sense! At first I thought it was using something from the `built_pkg` to determine the salt, but it looks like it's only using the package name for `contract_salt_map`. No worries then :)